pkg/controlplane/instance.go

```diff
@@ -493,6 +495,29 @@ func (c completedConfig) New(delegationTarget genericapiserver.DelegationTarget)
 		return nil
 	})
 
+	if PeerProxyFeatureEnabled() {
+		peerEndpointReconcilerCtrl := controlplane.New(controlplane.Config{
```

Comment:
huh... having a separate reconciler for this seems really weird to me... it means the peer IPs and list of active peer servers can drift/differ from the servers considered active/healthy and used as inputs to the kubernetes service endpoints. That can happen because the sync interval drifts, or because one of the reconcilers encounters errors the other one doesn't. Don't we want those to be in lockstep?